app/models/order_cycle.rb

```diff
@@ -250,6 +250,13 @@ def refresh_products_cache
     OpenFoodNetwork::ProductsCache.order_cycle_changed self
   end
 
+  def items_bought_by_user(user, distributor)
+    # The Spree::Order.complete scope only checks for completed_at date, does not ensure state is "complete"
+    orders = Spree::Order.complete.where(state: "complete", user_id: user, distributor_id: distributor, order_cycle_id: self)
+    items = orders.map(&:line_items).flatten
```

Comment:
Here you have almost all the available options http://blog.bigbinary.com/2013/07/01/preload-vs-eager-load-vs-joins-vs-includes.html

But yes, something like that would work.